functions.php

```diff
@@ -302,13 +302,29 @@ function render_related_posts_block(array $attributes): string
     // Encode the query attributes to JSON for the block template
     $query_json = wp_json_encode($attributes['query_attributes'], JSON_UNESCAPED_SLASHES);
 
+    // Dynamically render link to News & Stories page
+    $news_stories_url = '';
+    $news_stories_page = get_posts([
+        'title' => 'News & Stories',
+        'post_type' => 'page',
+    ]);
+
+    if ($news_stories_page) {
+        $id = $news_stories_page[0];
+        $id = $id->ID;
+        $news_stories_url = get_permalink($id);
+    }
+
     // Define the HTML output for the block
     $output = '<!-- wp:query ' . $query_json . ' -->
             <div class="wp-block-query posts-list p4-query-loop is-custom-layout-list"><!-- wp:group {"layout":{"type":"flex","justifyContent":"space-between"}} -->
             <div class="wp-block-group related-posts-block"><!-- wp:heading {"lock":{"move":true}} -->
             <h2 class="wp-block-heading">' . __('Related Posts', 'planet4-blocks') . '</h2>
             <!-- /wp:heading -->
-            <!-- wp:navigation-link {"label":"' . __('See all posts', 'planet4-blocks') . '","url":"http://www.planet4.test/news-stories/","className":"see-all-link"} /-->
+            ' . (!empty($news_stories_url) ?
```

Comment:
Ideally we would create this chunk of code as a reusable variable, to avoid duplication